tools/pylint-extensions/pylint-guidelines-checker/pylint_guidelines_checker.py

```diff
@@ -1985,6 +1960,185 @@ def visit_importfrom(self, node):
+
+            for i in node.body:
+                try:
+                    if len(i.name) > self.STANDARD_CHARACTER_LENGTH:
+                        self.add_message(
+                            msgid=f"name-too-long",
+                            node=i,
+                            confidence=None,
+                        )        
+                except:
+                    # Gets the names of ast.Assign statements
+                    for j in i.targets:
+                        if len(j.name) > self.STANDARD_CHARACTER_LENGTH:
+                            self.add_message(
+                                msgid=f"name-too-long",
+                                node=i,
+                                confidence=None,
+                            )          
+        except:
+            pass      
+
+    visit_asyncfunctiondef = visit_functiondef
+
+
+class DeleteOperationReturnStatement(BaseChecker):
+    __implements__ = IAstroidChecker
+
+    """Rule to check that a delete* or begin_delete* returns None or LROPoller[None]."""
+    name = "delete-operation-wrong-return-type"
+    priority = -1
+    msgs = {
+        "C4752": (
+            "delete* or begin_delete* does not return None or LROPoller[None].",
+            "delete-operation-wrong-return-type",
+            "delete* or begin_delete* functions should return None or LROPoller[None]."
+        ),
+    }
+
+    def visit_functiondef(self,node):
+        """Visits all delete functions and checks that their return types
+        are LROPoller or None. """
+        try:
+            if node.name.startswith("delete") or node.name.startswith("begin_delete"):
+                inferred = node.infer_call_result()
+                for x in inferred:
+                    if str(x)=="Uninferable" or x == None or x.value == None or \
+                        x.name == "LROPoller" or x.name == "AsyncLROPoller":
+                            return
+                self.add_message(
+                    msgid=f"delete-operation-wrong-return-type",
+                    node=node,
+                    confidence=None,
+                )   
+        except:
+            pass
+    
+    visit_asyncfunctiondef = visit_functiondef
+
+
+class AsyncMethodsReturnAsyncIterables(BaseChecker):
+    __implements__ = IAstroidChecker
+
+    """Rule to check that an async method returns an async iterable."""
+    name = "async-return-async-iterable"
+    priority = -1
+    msgs = {
+        "C4753": (
+            "async method returns sync iterable.",
+            "async-return-async-iterable",
+            "async methods should return async iterables."
+        ),
+    }
+
+    def visit_asyncfunctiondef(self, node):
+        """Checks all async functions, if there is a return statement, checks that it is preceded by an await. 
+        If it is a call to a function, it checks that the function is async. i.e AsyncLROPoller"""
+        try:
+            if "Client" in node.parent.name:
```

Comment:
Not sure I understand what this one should be checking. From the description it seems like we should we be looking for an ItemPaged returned on an async method?